libs/shared/src/features/audio-playback/audio-playback.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, ViewChild, ElementRef, OnDestroy, Input } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { AudioService } from './audio-playback.service';
+import { MatIconModule } from '@angular/material/icon';
+
+@Component({
+  selector: 'picsa-audio-playback',
+  standalone: true,
+  imports:[CommonModule, MatIconModule],
+  templateUrl: './audio-playback.component.html',
+  styleUrls: ['./audio-playback.component.scss'],
+})
+export class AudioPlaybackComponent implements OnDestroy {
+  @Input() audioUrl: string;
+  isAudioPlaying = false;
+
+  constructor(private audioService: AudioService) {}
+
+  togglePlayback(): void {
+    if (this.audioService.isPlaying()) {
+      this.audioService.pauseAudio();
+      this.isAudioPlaying = true;
+    } else {
+      this.audioService.playAudio(this.audioUrl);
+      this.isAudioPlaying = false;
+    }
+  }
+
+  isPlaying(): boolean {
+    return this.audioService.isPlaying();
```

Comment:
nit(non-blocking): I like how you have linked the audio play detection to the service, as it's possible that the service will stop component playback (e.g. if the user tries to start a second audio while the first one is playing), however you might need to remove the `isAudioPlaying` variable from this component to avoid confusion